tests/ops/api/v1/endpoints/test_messaging_endpoints.py

```diff
@@ -0,0 +1,761 @@
+import json
+
+import pytest
+from fastapi_pagination import Params
+from sqlalchemy.orm import Session
+from starlette.testclient import TestClient
+
+from fides.api.ops.api.v1.scope_registry import (
+    MESSAGING_CREATE_OR_UPDATE,
+    MESSAGING_DELETE,
+    MESSAGING_READ,
+)
+from fides.api.ops.api.v1.urn_registry import (
+    MESSAGING_BY_KEY,
+    MESSAGING_CONFIG,
+    MESSAGING_SECRETS,
+    V1_URL_PREFIX,
+)
+from fides.api.ops.models.messaging import MessagingConfig
+from fides.api.ops.schemas.messaging.messaging import (
+    MessagingServiceDetails,
+    MessagingServiceSecrets,
+    MessagingServiceType,
+)
+
+PAGE_SIZE = Params().size
+
+
+class TestPostMessagingConfig:
+    @pytest.fixture(scope="function")
+    def url(self) -> str:
+        return V1_URL_PREFIX + MESSAGING_CONFIG
+
+    @pytest.fixture(scope="function")
+    def payload(self):
+        return {
+            "name": "mailgun",
+            "service_type": MessagingServiceType.MAILGUN.value,
+            "details": {MessagingServiceDetails.DOMAIN.value: "my.mailgun.domain"},
+        }
+
+    @pytest.fixture(scope="function")
+    def payload_twilio_email(self):
+        return {
+            "name": "twilio_email",
+            "service_type": MessagingServiceType.TWILIO_EMAIL.value,
+        }
+
+    @pytest.fixture(scope="function")
+    def payload_twilio_sms(self):
+        return {
+            "name": "twilio_sms",
+            "service_type": MessagingServiceType.TWILIO_TEXT.value,
+        }
+
+    def test_post_email_config_not_authenticated(
+        self, api_client: TestClient, payload, url
+    ):
+        response = api_client.post(url, headers={}, json=payload)
+        assert 401 == response.status_code
+
+    def test_post_email_config_incorrect_scope(
+        self,
+        api_client: TestClient,
+        payload,
+        url,
+        generate_auth_header,
+    ):
+        auth_header = generate_auth_header([MESSAGING_READ])
+        response = api_client.post(url, headers=auth_header, json=payload)
+        assert 403 == response.status_code
+
+    def test_post_email_config_with_invalid_mailgun_details(
+        self,
+        db: Session,
+        api_client: TestClient,
+        url,
+        payload,
+        generate_auth_header,
+    ):
+        payload["details"] = {"invalid": "invalid"}
+
+        auth_header = generate_auth_header([MESSAGING_CREATE_OR_UPDATE])
+        response = api_client.post(url, headers=auth_header, json=payload)
+        assert 422 == response.status_code
+        assert json.loads(response.text)["detail"][0]["msg"] == "field required"
+        assert (
+            json.loads(response.text)["detail"][1]["msg"]
+            == "extra fields not permitted"
+        )
```

Comment:
All of the `json.loads(response.text)` in these tests can be simplified to `resonse.json()`

```suggestion
        assert response.json()["detail"][0]["msg"] == "field required"
        assert (
            response.json()["detail"][1]["msg"]
            == "extra fields not permitted"
        )
```

Reply:
Nice, thanks!